src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs

```diff
@@ -45,20 +44,21 @@ public static async Task LogAsync(IEnumerable<IRestoreTargetGraph> graphs, Remot
                     messageTasks.Add(GetMessageAsync(
                         ddi.Framework.ToString(),
                         unresolved,
-                        remoteLibraryProviders, sourceCacheContext,
-                        logger,
+                        context.FilterDependencyProvidersForLibrary(unresolved),
+                        context.CacheContext,
+                        context.Logger,
                         token));
                 }
             }
 
             var messages = await Task.WhenAll(messageTasks);
-            await logger.LogMessagesAsync(DiagnosticUtility.MergeOnTargetGraph(messages));
+            await context.Logger.LogMessagesAsync(DiagnosticUtility.MergeOnTargetGraph(messages));
```

Comment:
A reference to `RemoteWalkContext.Logger` was passed earlier. In this PR I changed the method signature to accept `RemoteWalkContext` and hence modified code to access `logger` directly from the `context`